Client/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp

```diff
@@ -1301,6 +1303,53 @@ int CLuaElementDefs::IsElementLocal(lua_State* luaVM)
     return 1;
 }
 
+int CLuaElementDefs::IsElementData(lua_State* luaVM)
+{
+    //  bool isElementData ( element theElement, string key [ , bool inherit = true ] )
+    CClientEntity* pEntity;
+    SString        strKey;
+    bool           bInherit;
+
+    CScriptArgReader argStream(luaVM);
+    argStream.ReadUserData(pEntity);
+    argStream.ReadString(strKey);
+    argStream.ReadBool(bInherit, true);
+    
+    if (argStream.HasErrors())
+    {
+        return luaL_error(luaVM, argStream.GetFullErrorMessage());
+    }
+
+    if (strKey.length() > MAX_CUSTOMDATA_NAME_LENGTH)
+    {
+        // Warn and truncate if key is too long
+        m_pScriptDebugging->LogCustom(luaVM, SString("Truncated argument @ '%s' [%s]", lua_tostring(luaVM, lua_upvalueindex(1)),
+                                                     *SString("string length reduced to %d characters at argument 2", MAX_CUSTOMDATA_NAME_LENGTH)));
+        strKey = strKey.Left(MAX_CUSTOMDATA_NAME_LENGTH);
+    }
+
+    CLuaMain* pLuaMain = m_pLuaManager->GetVirtualMachine(luaVM);
+    if (pLuaMain)
+    {
+        // Look for data with given key
+        CLuaArgument* pVariable = pEntity->GetCustomData(strKey, bInherit);
+        if (pVariable)
+        {
+            lua_pushboolean(luaVM, true);
+            return 1;
+        }
+        else
+        {
+            lua_pushboolean(luaVM, false);
+            return 1;
+        }
```

Comment:
You can replace all this mess with:
```cpp
lua_pushboolean(luaVM, pEntity->GetCustomData(strKey, bInherit) != NULL);
```

Also, maybe move the key length check after checking for **pLuaMain**.